src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1040,6 +1040,19 @@ internal void OnAdditionalDocumentTextUpdatedOnDisk(DocumentId documentId)
             this.OnAdditionalDocumentTextLoaderChanged(documentId, vsDoc.Loader);
         }
 
+        internal void OnIsComplete(ProjectId projectId, bool succeeded)
+        {
+            // this can't be null
+            var state = CurrentSolution.GetProjectState(projectId);
+            Contract.ThrowIfNull(state);
+
+            // if state is different than what we have
+            if (succeeded != state.ProjectInfo.IsComplete)
+            {
+                OnProjectReloaded(state.ProjectInfo.WithIsComplete(succeeded));
```

Comment:
so i think i hate the 'IsComplete' name.  What is this really trying to convey?  That we were able to successfully load all references that this project has?  If so, why not call it "HasSuccessfullyLoadedAllReferences".  And, if we need to convey the difference between this and dependent projects we could have:

HasSuccessfullyLoadedAllDirectReferences and HasSuccessfullyLoadedAllTransitiveReferences.

'Complete' doesn't mean anything to me, and i think we need better names to make it clear what's going on.